### PR TITLE
Microsoft365defender - Set app_id to support creds

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] Ready

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/Microsoft365Defender/ReleaseNotes/2_0_4.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Microsoft 365 Defender (Beta)
+- Added the *Application ID* integration parameter to support credentials fetching object.
```
Comment (dantavori) [inline Packs/Microsoft365Defender/ReleaseNotes/2_0_4.md:4]:
```suggestion
- Added the *Client Secret* integration parameter to support credentials fetching object.
```

File: Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Pipfile [hunk 2]
```diff
@@ -13,6 +13,8 @@ pytest-asyncio = "*"
 [packages]
 pytest = "*"
 requests = "*"
+cryptography = "*"
```
Comment (dantavori) [inline Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Pipfile:16]: was this added to make the unittests pass?

File: Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py [hunk 3]
```diff
@@ -530,12 +530,15 @@ def main() -> None:
...
     proxy = params.get('proxy', False)
-    app_id = params.get('app_id')
+    app_id = params.get('app_id') or params.get('_app_id')
     base_url = params.get('base_url')
 
-    tenant_id = params.get('tenant_id')
+    tenant_id = params.get('tenant_id') or params.get('_tenant_id')
     client_credentials = params.get('client_credentials', False)
-    enc_key = params.get('enc_key')
+    enc_key = params.get('enc_key') or (demisto.getParam('credentials') or {}).get('password')
```
Comment (dantavori) [inline Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py:538]:
```suggestion
    enc_key = params.get('enc_key') or (params.get('credentials') or {}).get('password')
```

File: Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py [hunk 4]
```diff
@@ -530,12 +530,15 @@ def main() -> None:
...
+
+    if not app_id:
+        raise Exception('Aplication ID must be provided.')
```
Comment (dantavori) [inline Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py:541]: what about tenant_id and enc_key?

**Review by dantavori — CHANGES_REQUESTED** [review]

File: Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py (hunk 4 above)
Comment (dantavori) [reply Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py:541]: btw, let's put all of this inside the try block

File: Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Pipfile (hunk 2 above)
Comment (yaakovpraisler) [reply Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Pipfile]: removed

File: Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py (hunk 4 above)
Comment (yaakovpraisler) [reply Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py:541]: The tenant_id and enc_key are not required if the user used the device code authentication flow.

**Review by dantavori — APPROVED** [review]

File: Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py [hunk 5]
```diff
@@ -546,6 +546,9 @@ def main() -> None:
     args = demisto.args()
 
     try:
+        if not app_id:
+            raise Exception('Aplication ID must be provided.')
```
Comment (dantavori) [inline Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py:550]:
```suggestion
            raise Exception('Application ID must be provided.')
```

**Pull request merged**